src/test/kotlin/com/zepben/evolve/database/paths/LocalEwbDataFilePathsTest.kt

```diff
@@ -246,41 +243,51 @@ class EwbDataFilePathsTest {
         val emptyDirectories = listOf("2555-11-11")
 
-        val allDirectories = (hasNetworkDirectories + emptyDirectories)
-
-        every { listFiles(baseDir) } answers { allDirectories.map { Paths.get(baseDir.toString(), it) }.iterator() }
-
-        every { exists(any()) } returns false
-        hasNetworkDirectories.forEach {
-            every { exists(baseDir.datedPath(LocalDate.parse(it), DatabaseType.NETWORK_MODEL.fileDescriptor)) } returns true
-        }
+        every { listFiles(baseDir) } answers {
+            (hasNetworkDirectories.map { Paths.get(baseDir.toString(), it, "${it}-network-model.sqlite") } +
+                emptyDirectories.map { Paths.get(baseDir.toString(), it) }).iterator() }
 
         assertThat(ewbPaths.getNetworkModelDatabases(), equalTo(hasNetworkDirectories.map { LocalDate.parse(it) }))
 
-        verifySequence {
-            listFiles(baseDir)
-            allDirectories.forEach {
-                isDirectory(Paths.get(baseDir.toString(), it))
-                exists(baseDir.datedPath(LocalDate.parse(it), DatabaseType.NETWORK_MODEL.fileDescriptor))
-            }
-        }
+        verify { listFiles(baseDir) }
     }
 
-    private fun validateClosest(expectedDate: LocalDate?, expectedExistCalls: Int, searchForwards: Boolean = false) {
+    private fun validateClosest(expectedDate: LocalDate?, searchForwards: Boolean = false) {
         DatabaseType.entries.filter { it.perDate }.forEach {
-            validateClosest(it, expectedDate, expectedExistCalls, searchForwards)
+            validateClosest(it, expectedDate, searchForwards)
         }
     }
 
-    private fun validateClosest(type: DatabaseType, expectedDate: LocalDate?, expectedExistCalls: Int, searchForwards: Boolean = false) {
-        // Clear the calls, so we are not affected by previous calls that may not have been validated.
-        clearMocks(exists, answers = false)
-
+    private fun validateClosest(type: DatabaseType, expectedDate: LocalDate?, searchForwards: Boolean = false) {
         assertThat(ewbPaths.findClosest(type, 10, today, searchForwards), equalTo(expectedDate))
-
-        verify(exactly = expectedExistCalls) { exists(any()) }
     }
 
     private fun Path.datedPath(date: LocalDate, name: String): Path =
         resolve(date.toString()).resolve("$date-$name.sqlite")
+
+    @Test
+    internal fun enumerateDescendants() {
+        val descendants = listOf(
+            Path.of(today.toString(), "${today}-network-model.sqlite"),
+            Path.of(today.toString(), "${today}-customer.sqlite"),
+            Path.of("results-cache.sqlite"),
+            Path.of("weather-readings.sqlite")
+        )
+
+        every { listFiles(baseDir) } answers { descendants.iterator() }
+
+        val result = ewbPaths.enumerateDescendants()
+        assertThat(result.asSequence().count(), `is`(descendants.size))
+        ewbPaths.enumerateDescendants().forEach {
+            assertTrue(descendants.contains(it))
```

Comment:
todo: Use Hamcrest `assertThat` rather than `junit`. Should be in the form of `assertThat("$it - all listed files should havebeen found in the results.", descendants.contains(it))`